src/main/java/org/cryptomator/ui/error/ErrorController.java

```diff
@@ -95,13 +127,141 @@ public void copyDetails() {
+	 * - A negative value (-1) if ed1 has an exact match with the error code in the title and ed2 does not have a match,
+	 * - A positive value (1) if ed1 does not have a match and ed2 has an exact match with the error code in the title,
+	 * - Or 0 if both ErrorDiscussion objects either have an exact match or do not have a match with the error code in the titles.
+	 */
+	public int compareExactMatch(ErrorDiscussion ed1, ErrorDiscussion ed2) {
+		if (ed1.title.contains(getErrorCode()) && !ed2.title.contains(getErrorCode())) {
+			return -1;
+		} else if (!ed1.title.contains(getErrorCode()) && ed2.title.contains(getErrorCode())) {
+			return 1;
+		} else {
+			return 0;
+		}
+	}
+
+	/**
+	 * Compares two ErrorDiscussion objects based on the presence of a second-level match with the error code in their titles and returns the result.
+	 *
+	 * @param ed1 The first ErrorDiscussion object.
+	 * @param ed2 The second ErrorDiscussion object.
+	 * @return An integer indicating the comparison result based on the presence of a second-level match with the error code in the titles:
+	 * - A negative value (-1) if ed1 has a second-level match with the error code in the title and ed2 does not have a match,
+	 * - A positive value (1) if ed1 does not have a match and ed2 has a second-level match with the error code in the title,
+	 * - Or 0 if both ErrorDiscussion objects either have a second-level match or do not have a match with the error code in the titles.
+	 */
+	public int compareSecondLevelMatch(ErrorDiscussion ed1, ErrorDiscussion ed2) {
+		String value = " " + errorCode.methodCode() + ErrorCode.DELIM + errorCode.rootCauseCode();
+		if (ed1.title.contains(value) && !ed2.title.contains(value)) {
+			return -1;
+		} else if (!ed1.title.contains(value) && ed2.title.contains(value)) {
+			return 1;
+		} else {
+			return 0;
+		}
+	}
 
+	/**
+	 * Compares two ErrorDiscussion objects based on the presence of a third-level match with the error code in their titles and returns the result.
+	 *
+	 * @param ed1 The first ErrorDiscussion object.
+	 * @param ed2 The second ErrorDiscussion object.
+	 * @return An integer indicating the comparison result based on the presence of a third-level match with the error code in the titles:
+	 * - A negative value (-1) if ed1 has a third-level match with the error code in the title and ed2 does not have a match,
+	 * - A positive value (1) if ed1 does not have a match and ed2 has a third-level match with the error code in the title,
+	 * - Or 0 if both ErrorDiscussion objects either have a third-level match or do not have a match with the error code in the titles.
+	 */
+	public int compareThirdLevelMatch(ErrorDiscussion ed1, ErrorDiscussion ed2) {
+		String value = " " + errorCode.methodCode();
+		if (ed1.title.contains(value) && !ed2.title.contains(value)) {
+			return -1;
+		} else if (!ed1.title.contains(value) && ed2.title.contains(value)) {
+			return 1;
+		} else {
+			return 0;
+		}
+	}
```

Comment:
Is this really needed? If we do a `filter(this::isPartialMatchFilter)` before sorting the stream, we can be sure all discussions contain the `methodCode`, so this always returns `0`, i.e. not affecting the order.

Reply:
was no longer needed and was removed